src/kaltura-live-plugin.tsx

```diff
@@ -78,19 +79,61 @@ export class KalturaLivePlugin
                 ks: playerConfig.provider.ks
             });
         }
+
+        _player.addEventListener(
+          _player.Event.SOURCE_SELECTED,
+          this._activatePlugin
+        );
     }
 
     onPluginSetup(): void {}
 
     onMediaLoad(): void {
-        this._isEntryLiveType();
+      if (this.isMediaLive) {
+        this.isMediaLoaded = true;
+        this.updateLiveTag();
+        this._player.addEventListener(this._player.Event.FIRST_PLAY, this._handleFirstPlay);
```

Comment:
you can use eventManager from playkit to listen and remove listener it'll be easier for you to handle